tests/SpatialOperationsSpec.hs

```diff
@@ -66,3 +71,35 @@ spatialOpsSpecs = describe "Tests Contains" $ do
     (ensurePoint $ envelope point) `shouldBe` point
     (ensureMultiLineString $ boundary poly2) `shouldBe` env2
     convexHull poly2 `shouldBe` env3
+
+  it "can use STRTrees" $ do
+    let points = makePointGeo <$> [(0.1,0.1), (0.9, 0.9)]
+        polygon = makePolygonGeo [[(0,0),(0,1),(1,1),(1,0),(0,0)]]
+        tree = STR.fromList $ zip points [(0::Int)..]
+        result = STR.lookup polygon tree
+    result `shouldBe` V.fromList [0,1]
+
+  it "can run STRTrees on larger data" $ do
+    points <- (fmap ensurePoint) <$> loadThingsFromFile "tests/sampledata/points.csv"
+    polygons <- (fmap ensurePolygon) <$> loadThingsFromFile "tests/sampledata/polygons.csv"
+
+    -- The following line takes a long time to run (>20 seconds) but gives us a real baseline count
+    -- count <- timeIt $! layerIntersect polygons points
+    -- count `shouldBe` 110
+    let count = 110
+    count' <- timeIt $ layerIntersectSTR polygons points
+    count' `shouldBe` count
+
+layerIntersect :: [Geometry Polygon] -> [Geometry Point] -> IO Int
+layerIntersect polygons points = do
+  let things = sum $ fmap (f points) polygons
+  if (things > 0) then pure things else pure 0
+  where f points polygon = length $ filter (\p -> contains polygon p) points
+
+layerIntersectSTR :: [Geometry Polygon] -> [Geometry Point] -> IO Int
```

Comment:
I've included the actual implementation of layer intersection using the STR tree here as it's not obvious that a query isn't the same as an intersect/contains query... but the numbers line up with the dumb impl so... thats something anyways :)